Guidance on UUIDs in POST/PUT

kperk103 opened the pull request:
This will add documentation on handling UUIDs in POST/PUT.

Closes #51.

Commit:
Add documentation on handling UUIDs

This adds documentation on handling UUIDs in `PUT` and
`PATCH`, including which circumstances each operation
should be used for.

Commit:
Modify `POST` and `PUT` descriptions

This edits the descriptions of `POST` and `PUT` to sound better.
The content is all still the same.

schen2166 commented on docs/handling-uuids-on-post-put.md:
```diff
@@ -0,0 +1,53 @@
+ error will be returned. If an object with the specified UUID exists, the `PUT` request will
+ update the existing object. If no objects exist with the specified UUID, a new object will
+ be created with that UUID.
+
```
Suggestion: write these as bullets for better clarity. Would be also be easier to read
If UUID is specified in both the path and body...
If an object with the specified UUID exists...
If no objects exist with the specified UUID...

laurelmay replied:
I think we could solve this by keeping the narrative and just moving the relevant diagram to this location

kperk103 replied:
@schen2166 I chose not to write it as bullets because it appears that almost all of the documentation written in the repository is written in paragraphs so I wanted to keep it consistent with that. @kylelaker I agree. The diagrams would help reinforce what each section is saying.

Commit:
Move diagrams for clarity

The diagrams have been moved to the relevant section to reinforce
what each section is saying.

Commit:
Fix file to pass Lint

Removes an extra space to pass the Lint check

tuckerzp commented on docs/handling-uuids-on-post-put.md:
```diff
@@ -0,0 +1,51 @@
+
+When creating new objects, `POST` is the preferred option. The UUID of the new object
+does not need to be specified, so `POST` should be used when there is no UUID preference.
+If the UUID is specified in the body, an HTTP 4xx error will be returned.
```
Isn't the error now specified by #54?

kperk103 replied:
I'm not sure if 409 would be the response to the UUIDs not matching. It is returned when there is a request conflict with the current state (the most common example being an issue with versioning) so I'm not sure if mismatched UUIDs would fall under this or not

tuckerzp replied:
Ok that makes sense. If it is not a `409` error, is there one more applicable? I don't know if we want just `4xx`

laurelmay replied:
Mismatched UUIDs is what we're talking about primarily at the moment with `409 Conflict` status code. That is the primary conflict in play

tuckerzp reviewed (changes requested):
Looks great so far! I just had a quick question about the error specified.

tuckerzp committed:
Merge branch 'develop' into feature/guidance-on-post-put

laurelmay commented:
Can you add a link to this from the `openapi.yaml` file?

Commit:
Add link to guidance in openapi

A link to the guidance on handling uuids in post/put is added
in openapi.yaml. Additionally, it specifies that a 409
error is returned if the uuids are mismatched.

Commit:
Merge branch 'feature/guidance-on-post-put' of github.com:EasyDynamics/oscal-rest into feature/guidance-on-post-put
